lib/active_model/serializer.rb

```diff
@@ -27,6 +27,20 @@ class Serializer
       )
     /x
 
+    # Hashes contents of file for +_cache_digest+
+    def self.digest_caller_file(caller_line)
+      serializer_file_path = caller_line[CALLER_FILE]
+      serializer_file_contents = IO.read(serializer_file_path)
+      Digest::MD5.hexdigest(serializer_file_contents)
+    rescue TypeError, Errno::ENOENT
+      warn <<-EOF.strip_heredoc
+       Cannot digest non-existent file: '#{caller_line}'.
+       Please set `::_cache_digest` of the serializer
+       if you'd like to cache it.
+       EOF
+      ''.freeze
```

Comment:
Ok

B mobile phone

> On Oct 21, 2015, at 6:20 PM, Lucas Hosseini <notifications@github.com> wrote:
> 
> In lib/active_model/serializer.rb:
> 
> > @@ -27,6 +27,20 @@ class Serializer
> >        )
> >      /x
> >  
> > +    # Hashes contents of file for +_cache_digest+
> > +    def self.digest_caller_file(caller_line)
> > +      serializer_file_path = caller_line[CALLER_FILE]
> > +      serializer_file_contents = IO.read(serializer_file_path)
> > +      Digest::MD5.hexdigest(serializer_file_contents)
> > +    rescue TypeError, Errno::ENOENT
> > +      warn <<-EOF.strip_heredoc
> > +       Cannot digest non-existent file: '#{caller_line}'.
> > +       Please set `::_cache_digest` of the serializer
> > +       if you'd like to cache it.
> > +       EOF
> > +      ''.freeze
> Yeah but what I meant was that the bug actually arose in the case the source file was deemed non-existent because the regex would not allow for whitespace.
> 
> —
> Reply to this email directly or view it on GitHub.
>